docker_auth_test.go

```diff
@@ -0,0 +1,104 @@
+package testcontainers
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const exampleAuth = "https://example-auth.com"
+const indexDockerIO = "https://index.docker.io/v1/"
+
+var testDockerConfigDirPath = filepath.Join("testresources", ".docker")
+
+func TestGetDockerConfig(t *testing.T) {
+	t.Run("without DOCKER_CONFIG env var retrieves default", func(t *testing.T) {
```

Comment:
This test assumes that the default file as written by Docker is present, correct? I wonder if it is worth protecting this environment assumption with an early assert, so we would understand through this assert failing, that the environment failed the assumption (which is something else from our code failing).

Reply:
Do you mean checking that the file does exist? This is what L19:20 does

Reply:
Oh I understand now: you mean checking that the default file does exist in the scope of the test, right? I've added the check here: https://github.com/testcontainers/testcontainers-go/pull/869/commits/402471b8ac6f1095fb2cdcbb7b917dbf8b7e5f1c